vlm/src/main/scala/geotrellis/contrib/vlm/package.scala

```diff
@@ -69,6 +75,118 @@ package object vlm {
     new StreamingByteReader(rr)
   }
 
+  implicit class tokenMethods(val token: Long) extends AnyVal {
+
+    def getProjection(): Option[String] = {
+      val crs = Array.ofDim[Byte](1 << 16)
+      GDALWarp.get_crs_wkt(token, 0, crs)
+      Some(new String(crs, "UTF-8"))
+    }
+
+    def rasterExtent(): RasterExtent = {
+      val transform = Array.ofDim[Double](6)
+      val width_height = Array.ofDim[Int](2)
+      GDALWarp.get_transform(token, 0, transform)
+      GDALWarp.get_width_height(token, 0, width_height)
+
+      val x1 = transform(0)
+      val y1 = transform(3)
+      val x2 = x1 + transform(1) * width_height(0)
+      val y2 = y1 + transform(5) * width_height(1)
+      val e = Extent(
+        math.min(x1, x2),
+        math.min(y1, y2),
+        math.max(x1, x2),
+        math.max(y1, y2))
+
+      RasterExtent(e,
+        math.abs(transform(1)), math.abs(transform(5)),
+        width_height(0), width_height(1))
+    }
+
+    def resolutions: List[RasterExtent] = {
+      val N = 1 << 8;
+      val widths = Array.ofDim[Int](N)
+      val heights = Array.ofDim[Int](N)
+      val extent = token.extent
+      GDALWarp.get_overview_widths_heights(token, 0, widths, heights)
+      widths.zip(heights).flatMap({ case (w, h) =>
+        if (w > 0 && h > 0) Some(RasterExtent(extent, cols = w, rows = h))
+        else None
+      }).toList
+    }
+
+    def extent(): Extent = token.rasterExtent.extent
+
+    def bandCount(): Int = {
+      val count = Array.ofDim[Int](1)
+      GDALWarp.get_band_count(token, 0, count)
+      count(0)
+    }
+
+    def crs(): CRS = {
+      val crs = Array.ofDim[Byte](1 << 16)
```

Comment:
There is again a large array allocation ):

Reply:
Alot of what you see here is stub code.